scripts/commit_retriever.py

```diff
@@ -52,7 +52,18 @@ def retrieve_commits(self):
         hg.pull(update=True)
         hg.close()
 
-        repository.download_commits(self.repo_dir)
+        db.download_version(repository.COMMITS_DB)
+        if not db.is_old_version(repository.COMMITS_DB):
+            db.download(repository.COMMITS_DB, support_files_too=True)
+
+            for commit in repository.get_commits():
```

Comment:
nit: this line looks pretty strange... maybe you could add a method in repo to just get last commit